services/app-api/src/postgres/contractAndRates/findAllRatesStripped.test.ts

```diff
@@ -168,25 +138,68 @@ it('returns all rates with stripped down data', async () => {
     const rateBFromArrayAgain = strippedRatesAgain.find(
         (rate) => rate.id === rateBID
     )
+
+    if (!rateBFromArrayAgain) {
+        throw new Error('Expected to find rateB')
+    }
+
     // expect rateB to have expected data after undo withdraw rate
-    consoleLogFullData(rateBFromArrayAgain)
     expect(
-        rateBFromArrayAgain?.latestSubmittedRevision.submitInfo?.updatedReason
+        rateBFromArrayAgain.latestSubmittedRevision.submitInfo?.updatedReason
     ).toContain('Undo withdraw rateB')
-    expect(rateBFromArrayAgain?.initiallySubmittedAt).toStrictEqual(
+    expect(rateBFromArrayAgain.initiallySubmittedAt).toStrictEqual(
         submittedContractB.initiallySubmittedAt
     )
-    expect(rateBFromArrayAgain?.parentContractID).toEqual(submittedContractB.id)
-    expect(rateBFromArrayAgain?.status).toBe('RESUBMITTED')
-    expect(rateBFromArrayAgain?.consolidatedStatus).toBe('RESUBMITTED')
+    expect(rateBFromArrayAgain.parentContractID).toEqual(submittedContractB.id)
+    expect(rateBFromArrayAgain.status).toBe('RESUBMITTED')
+    expect(rateBFromArrayAgain.consolidatedStatus).toBe('RESUBMITTED')
     expect(
-        rateBFromArrayAgain?.latestSubmittedRevision?.unlockInfo?.updatedReason
+        rateBFromArrayAgain.latestSubmittedRevision?.unlockInfo?.updatedReason
     ).toContain('Undo withdraw rateB')
-    expect(rateBFromArrayAgain?.reviewStatusActions?.length).toBe(2)
-    expect(rateBFromArrayAgain?.reviewStatusActions?.[0].actionType).toBe(
+    expect(rateBFromArrayAgain.reviewStatusActions?.length).toBe(2)
+    expect(rateBFromArrayAgain.reviewStatusActions?.[0].actionType).toBe(
         'UNDER_REVIEW'
     )
-    expect(rateBFromArrayAgain?.reviewStatusActions?.[1].actionType).toBe(
+    expect(rateBFromArrayAgain.reviewStatusActions?.[1].actionType).toBe(
         'WITHDRAW'
     )
+
+    const selectedStrippedRates = must(
+        await findAllRatesStripped(client, {
+            rateIDs: [rateBFromArrayAgain.id],
+        })
+    )
+
+    // Expect only rate B to return
+    expect(selectedStrippedRates).toHaveLength(1)
+    expect(selectedStrippedRates[0].rateID).toBe(rateBFromArrayAgain.id)
+
+    const emptySelectedStrippedRatesOrError = must(
+        await findAllRatesStripped(client, { rateIDs: [] })
```

Comment:
question: Feels odd to be able to pass empty array of  `rateIDs` to a query filter that is for _returning only specific rates_ and then expect all them back still. I'd expect none to be returned with this entry. Why are we handling this way, was there a reason to do this? Are we thinking folks may pass in empty arrays accidentally?
 
  Seems like if you want just all the rates, you would call`findAllRatesStripped` without any arguments.